### DISCUSSION TITLE
Logging out for Passport-auth0

### DISCUSSION THREAD

**Discussion by christianto88** [blitz-js/blitz#2660] [Q&A] [answered]
I am implementing third party authentication using passport-auth0. Based on the example on the documentation, we just expose an api endpoint for login. What about the logout? Is there any function/api endpoint exposed?
The default logout Mutation that comes with blitz JS is just revoking the session.

**Comment by flybayer** [comment]
@christianto88 I know we had some discussion on this in Discord. What solution did you end up with?

**Comment by christianto88** [comment]
@flybayer I end up making GET request to Auth0 logout endpoint, providing client ID and redirectUrl as query params. That should work

**Comment by flybayer** [comment]
@christianto88 can you share your code snippet?

**Comment by christianto88** [comment, marked as answer]
```
<a
    href="#"
    onClick={async () => {
        await logoutMutation()
        try {
            router.replace(`https://${process.env.NEXT_PUBLIC_AUTH0_DOMAIN}/v2/logout?
            client_id=${process.env.NEXT_PUBLIC_AUTH0_CLIENT_ID}&returnTo=${process.env.RETURN_URL}`)
        } catch (error) {
            console.log(error)
        }
    }}
>
    Sign out
</a>
```
